examples/all_features/stringification.cpp

```diff
@@ -75,6 +75,33 @@ struct Foo
     friend bool operator==(const Foo&, const Foo&) { return false; }
 };
 
+struct MyStringType
+{
+    std::string s;
+    template <size_t N> MyStringType(const char (&in)[N]) : s(in, in+N) {}
+    friend bool operator==(const MyStringType &lhs, const MyStringType &rhs) { return lhs.s == rhs.s; }
+};
+
+template <typename T, typename K>
+std::ostream& operator<<(std::ostream& stream, const MyType<T, K>& in) {
+    stream << "[" << in.one << ", " << in.two << "]";
+    return stream;
+}
+
+// you can use the stream.write() method to write blocks of characters, and
+// you also can use a template operator<< if your code does not use
+// std::ostream and you'd like to avoid the include
+template <class OStream>
+OStream& operator<<(OStream& stream, const MyStringType &in) {
+    // the terminating \0 character may be present in size()
+    auto size = in.s.size() > 0u ? in.s.size() - 1u : 0u;
+    constexpr const char quote = '\'';
+    stream.write(&quote, 1u);
+    stream.write(in.s.data(), static_cast<unsigned>(size));
+    stream.write(&quote, 1u);
+    return stream;
+}
```

Comment:
**Here**. clang fails to compile when this function is a template. OTOH, if this function is not a template, then it compiles and does the right thing. VS2019 and gcc have no problem.

Also, to make it clear, stringification from the standard `CHECK_EQ()` call works as intended on every compiler - template or not.

This would seem a subtle SFINAE difference in behavior between `CHECK_EQ()` and both `INFO()`/`CHECK_MESSAGE()`.